mycity/mycity/test/intent_base_case.py

```diff
@@ -0,0 +1,56 @@
+import unittest
+
+import mycity.intents.intent_constants as intent_constants
+import mycity.mycity_controller as mcc
+import mycity.mycity_request_data_model as req
+import mycity.test.test_constants as test_constants
+
+
+
+
+###############################################################################
+# TestCase parent class for all intent TestCases, which are integration tests #
+# to see if any changes in codebase have broken response-request model.       #
+#                                                                             #
+# NOTE: Assumes that address has already been set.                            #
+###############################################################################
+
+class IntentBaseCase(unittest.TestCase):
+
+    __test__ = False
+
+    intent_to_test = None
```

Comment:
Subclasses are supposed to override both these fields but they don't. So, the test methods are called with intent_to_test = None and returns_reprompt_text = False.  I think setUpClass sets up the request.intent_name with the subclass's intent_to_test which is why this bug only comes up when the TestRunner decides whether to skip the reprompt_text tests